### DISCUSSION TITLE
Calibration crashing?

### DISCUSSION THREAD

**Discussion by Steiner-SE** [Lailloken/Exile-UI#250] [General]
I'm trying to set up the screen recognition for Delving, but getting nowhere as every attempt to calibrate seems to crash the script (or something).
I am on Win 11, I follow the instructions to line up the grid. When I select to calibrate the screen snippet tool appears, I drag a line along a long corridor as instructed and Windows says a screen snap was saved. That's the end of things for me.
After that I have a small white square saying "calibrating..." forever stuck on my screen Lailloken-UI vanishes and need to be restarted in order to be used again.
Am I doing something wrong?

(on a side note, when looking to buy gems, is the Omni key supposed to do anything, for me it does nothing)

**Comment by Lailloken** [comment]
> After that I have a small white square saying "calibrating..." forever stuck on my screen Lailloken-UI vanishes and need to be restarted in order to be used again

Maybe the area you screen-capped was too large and the CPU takes too long to analyze each individual pixel. Try it again with a smaller screen-cap.
<br>

> (on a side note, when looking to buy gems, is the Omni key supposed to do anything, for me it does nothing)

Only when the guide tells you to buy gems, it's not universal. The guide-panel has to be on screen, a prompt to buy gems has to be on it, and you have to be in a town or the Act 3 library.

**Reply by Steiner-SE** [reply]
> Maybe the area you screen-capped was too large and the CPU takes too long to analyze each individual pixel. Try it again with a smaller screen-cap.

Hmm, I thought I followed the Wiki guide to the letter and the area I marked was just a small string along the path as suggested. My computer is reasonably high end so computing power shouldn't be an issue (Ryzen 9 7950x with 32GB 6GHz DDR5).
Now that I look at the Snipet tool it defaults to window snap, I changed that to rectangle and will try, see if that was it.

> Only when the guide tells you to buy gems, it's not universal. The guide-panel has to be on screen, a prompt to buy gems has to be on it, and you have to be in a town or the Act 3 library.

Yes, the panel was open and showing a lot of gems to buy, using the Omni key does nothing for me. What I need to do is click each entry in the list one by one and the gem will be marked in the shop if it's there. (not a big issue, was just curious if it was supposed to work as I thought it would).

Thank you for your excellent tool.

**Reply by Lailloken** [reply]
That's a nice rig, I should have worded it differently. It might take quite a while until the CPU has brute-forced through all the pixels, while being held back by AHK's single-thread limitations.

Are you using the gear-tracker? The normal guide-panel is not clickable and cannot be used in the way you described.
![img](https://user-images.githubusercontent.com/61888437/184586843-10fe30cb-1825-4d77-8927-62eef5b67505.png)

Using the omni-key to highlight gems is tied to this panel and its prompts being on screen (and your character being in the aforementioned locations):
![image](https://user-images.githubusercontent.com/61888437/236668769-4d61dda3-2318-4f7d-89a5-50796ea81c85.png)

EDIT:
If you want to bulk-buy gems regardless of your guide-progress, you'll have to use the search-strings feature. Importing a guide from exile-leveling will automatically add an entry "hideout Lilly" with which to paste regex-strings for all gems in that build.

**Reply by Steiner-SE** [reply]
Well, seems the calibration issue was definitely operator error and overloading the poor script. The snippet tool defaulted to doing a full window screen snap, once I set it to a rectangular area snap the calibration finishes quickly stating it has sampled 2-3k pixels.
Still can't seem to get the tool to actually do anything for me so still have to figure things out.
Now I am calibrated and I've got the grid sizes set up and aligned. I left click a unconnected note and it tells me to scan first (as I always forget). I scan using the omni key and then left click the node again, and.. nothing? This is what I see (shouldn't there be green overlay colors for potential paths?)

![Untitled-1](https://user-images.githubusercontent.com/132791331/236702630-fae13116-26ef-409a-a7a3-66ab6ce31a87.jpg)

I do have an imported guide and use both the displays shown above, what I usually end up doing is just right click on each green entry at the vendor to search string them.

**Reply by Lailloken** [reply]
It's not reading the map correctly (it thinks every 'sector' has an intersection). There's probably too much incorrect pixel-data from calibration. By calibrating, you tell the script which pixel color-values represent paths on the map and what too look out for. The data should only contain shades of gray, but in your case it probably includes other, unrelated shades as well, which leads to incorrect readings.

You'll have to delete the data and re-do calibration. You can do it step by step, screen-capping one path after another and scanning the map in between to see how well it is picked up by the script.

I'll see if I can figure out the omni-key problems later.
